pilot/pkg/model/sidecar.go

```diff
@@ -517,9 +518,11 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 }
 
 func matchingServices(importedHosts []host.Name, service *Service, ilw *IstioEgressListenerWrapper) []*Service {
-	// If a listener is defined with port, we should match services with port.
-	// If an unix domain socket is given as a port, we should not match by port and include services based on hosts.
-	needsPortMatch := ilw.IstioListener != nil && ilw.IstioListener.Port != nil && ilw.IstioListener.Port.GetNumber() != 0
+	// If a listener is defined with port, we should match services with port except in the following cases.
+	//  - If an unix domain socket is given as a port, we should not match by port and include services based on hosts.
+	//  - If Port's protocol is proxy protocol in which case the egress listener is used as generic egress http proxy.
+	needsPortMatch := ilw.IstioListener != nil && ilw.IstioListener.Port != nil &&
+		ilw.IstioListener.Port.GetNumber() != 0 && ilw.IstioListener.Port.Protocol != proxyProtocol
```

Comment:
Why not add it to protocol list ?

Reply:
I am thinking of adding - just pushed to see what CI thinks